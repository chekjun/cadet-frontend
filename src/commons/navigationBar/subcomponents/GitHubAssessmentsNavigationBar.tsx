import {
  Alignment,
  Button,
  Classes,
  Icon,
  InputGroup,
  Menu,
  MenuItem,
  Navbar,
  NavbarGroup
} from '@blueprintjs/core';
import { IconNames } from '@blueprintjs/icons';
import { Popover2 } from '@blueprintjs/popover2';
import { Octokit } from '@octokit/rest';
import classNames from 'classnames';
import * as React from 'react';
import { NavLink } from 'react-router-dom';

import { ControlBarGitHubLoginButton } from '../../controlBar/ControlBarGitHubLoginButton';

type GitHubAssessmentsNavigationBarProps = DispatchProps & StateProps;

type DispatchProps = {
  handleGitHubLogIn: () => void;
  handleGitHubLogOut: () => void;
};

type StateProps = {
  octokit: Octokit;
  courses: string[];
  selectedCourse: string;
  setSelectedCourse: (course: string) => void;
  typeNames: string[];
};

/**
 * The white navbar for the website. Should only be displayed when using GitHub-hosted missions.
 *
 * @param props Component properties
 */
const GitHubAssessmentsNavigationBar: React.FC<GitHubAssessmentsNavigationBarProps> = props => {
  const handleClick = (e: any) => {
    handleChange(e);
  };

  const handleChange = (e: any) => {
    props.setSelectedCourse(e.target.innerText);
  };

  return (
    <Navbar className="NavigationBar secondary-navbar">
      <NavbarGroup align={Alignment.LEFT}>
        <NavLink
          to={`/githubassessments/missions`}
          activeClassName={Classes.ACTIVE}
          className={classNames('NavigationBar__link', Classes.BUTTON, Classes.MINIMAL)}
        >
          <Icon icon={IconNames.FLAME} />
          <div className="navbar-button-text hidden-xs hidden-sm">{props.typeNames[0]}</div>
        </NavLink>
      </NavbarGroup>

      <NavbarGroup align={Alignment.LEFT}>
        <NavLink
          to={`/githubassessments/${props.typeNames[1]}`}
          activeClassName={Classes.ACTIVE}
          className={classNames('NavigationBar__link', Classes.BUTTON, Classes.MINIMAL)}
        >
          <Icon icon={IconNames.FLAME} />
          <div className="navbar-button-text hidden-xs hidden-sm">{props.typeNames[1]}</div>
        </NavLink>
      </NavbarGroup>

      <NavbarGroup align={Alignment.LEFT}>
        <NavLink
          to={`/githubassessments/${props.typeNames[2]}`}
          activeClassName={Classes.ACTIVE}
          className={classNames('NavigationBar__link', Classes.BUTTON, Classes.MINIMAL)}
        >
          <Icon icon={IconNames.FLAME} />
          <div className="navbar-button-text hidden-xs hidden-sm">{props.typeNames[2]}</div>
        </NavLink>
      </NavbarGroup>

      <NavbarGroup align={Alignment.LEFT}>
        <NavLink
          to={`/githubassessments/${props.typeNames[3]}`}
          activeClassName={Classes.ACTIVE}
          className={classNames('NavigationBar__link', Classes.BUTTON, Classes.MINIMAL)}
        >
          <Icon icon={IconNames.FLAME} />
          <div className="navbar-button-text hidden-xs hidden-sm">{props.typeNames[3]}</div>
        </NavLink>
      </NavbarGroup>

      <NavbarGroup align={Alignment.LEFT}>
        <NavLink
          to={`/githubassessments/${props.typeNames[4]}`}
          activeClassName={Classes.ACTIVE}
          className={classNames('NavigationBar__link', Classes.BUTTON, Classes.MINIMAL)}
        >
          <Icon icon={IconNames.FLAME} />
          <div className="navbar-button-text hidden-xs hidden-sm">{props.typeNames[4]}</div>
        </NavLink>
      </NavbarGroup>


      <NavbarGroup align={Alignment.RIGHT}>
        <ControlBarGitHubLoginButton
          key="github"
          onClickLogIn={props.handleGitHubLogIn}
          onClickLogOut={props.handleGitHubLogOut}
        />
      </NavbarGroup>

      {props.octokit !== undefined && (
        <NavbarGroup align={Alignment.RIGHT}>
          <InputGroup
            disabled={true}
            leftElement={
              <Popover2
                content={
                  <Menu>
                    {props.courses.map((course: string) => (
                      <MenuItem key={course} onClick={handleClick} text={course} />
                    ))}
                  </Menu>
                }
                placement={'bottom-end'}
              >
                <Button minimal={true} rightIcon="caret-down" />
              </Popover2>
            }
            placeholder={'Select Course'}
            onChange={handleChange}
            value={props.selectedCourse}
          />
        </NavbarGroup>
      )}
    </Navbar>
  );
};

export default GitHubAssessmentsNavigationBar;
