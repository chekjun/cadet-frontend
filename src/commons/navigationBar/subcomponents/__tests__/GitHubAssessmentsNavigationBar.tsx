import { Octokit } from '@octokit/rest';
import { shallow } from 'enzyme';

import GitHubAssessmentsNavigationBar from '../GitHubAssessmentsNavigationBar';

const props = {
  handleGitHubLogIn: () => {},
  handleGitHubLogOut: () => {},
  octokit: new Octokit(undefined),
  courses: [],
  selectedCourse: '',
  setSelectedCourse: () => {},
  typeNames: []
};

test('Navbar renders correctly', () => {
  const navbar = <GitHubAssessmentsNavigationBar {...props} />;
  const tree = shallow(navbar);
  expect(tree.debug()).toMatchSnapshot();
});
