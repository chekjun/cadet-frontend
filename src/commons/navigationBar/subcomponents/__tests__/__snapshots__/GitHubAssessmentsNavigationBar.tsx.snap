// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders correctly 1`] = `
"<Blueprint3.Navbar className=\\"NavigationBar secondary-navbar\\">
  <Blueprint3.NavbarGroup align=\\"left\\">
    <NavLink to=\\"/githubassessments/missions\\" activeClassName=\\"bp3-active\\" className=\\"NavigationBar__link bp3-button bp3-minimal\\">
      <Blueprint3.Icon icon=\\"flame\\" />
      <div className=\\"navbar-button-text hidden-xs hidden-sm\\">
        Missions
      </div>
    </NavLink>
    <NavLink to=\\"/githubassessments/courses\\" activeClassName=\\"bp3-active\\" className=\\"NavigationBar__link bp3-button bp3-minimal\\">
      <Blueprint3.Icon icon=\\"flame\\" />
      <div className=\\"navbar-button-text hidden-xs hidden-sm\\">
        Courses
      </div>
    </NavLink>
  </Blueprint3.NavbarGroup>
  <Blueprint3.NavbarGroup align=\\"right\\">
    <ControlBarGitHubLoginButton loggedInAs={[undefined]} onClickLogIn={[Function: handleGitHubLogIn]} onClickLogOut={[Function: handleGitHubLogOut]} />
  </Blueprint3.NavbarGroup>
</Blueprint3.Navbar>"
`;
